### PR TITLE
Feat/add UI for confiming cbd imports

### PR DESCRIPTION
**Pull request by sergiomarrocoli**
We are pulling commitments from CBD. CFN does not want pledges to eat less meat under the supervision of mother to appear on the page. this adds an admin page where imported commitments can be quickly viewed and confirmed/rejected. 

Testing:
- run ``rails db:migrate:rest``` and ```rake import:commitments```
- sign in with test@test.com, which i snow an admin user
- should see an admin link in the nav drop down
- should see a list of the imported commitments
- accepting or rejecting a commitment should remove it, and it should then be viewable in the appropriate tab
- sign out and sign is as test2@test.com
- admin link should not be in navbar dropdown
- entering /admin/imported_commitments should send you to the home page

### CODE REVIEW

File: app/controllers/admin/imported_commitments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  before_action :user_is_admin
+  before_action :set_commitment, only: [:update]
+
+  def index
+    respond_to do |format|
+      format.json { 
+        @commitments = Commitment.where(commitment_source: 'cbd', cfn_approved: cfn_approved_param) 
+        render json: @commitments.as_json(only: [:id, :name, :description])
+      }
+      format.html {}
```
Comment (yndajas) [inline app/controllers/admin/imported_commitments_controller.rb:13]: Is this needed?

File: app/controllers/admin/imported_commitments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+class Admin::ImportedCommitmentsController < ActionController::Base
+  layout 'application'
+
+  before_action :user_is_admin
```
Comment (yndajas) [inline app/controllers/admin/imported_commitments_controller.rb:4]: Should this method/before action be verb-y like `authenticate_user` in the other controllers?

File: app/javascript/components/nav/NavDropdown.vue [hunk 3]
```diff
@@ -71,6 +78,16 @@ export default {
...
+
+    adminPath: {
+      type: String,
+      required: false,
+    },
+
+    adminText: {
+      type: String,
+      required: false,
+    },
```
Comment (yndajas) [inline app/javascript/components/nav/NavDropdown.vue:90]: More conventional to use `default: ''` for both of these props

File: test/fixtures/users.yml [hunk 4]
```diff
@@ -4,4 +4,9 @@ user_1:
...
 user_2:
   email: test2@test.com
-  confirmed_at: <%= Time.now %>
\ No newline at end of file
+  confirmed_at: <%= Time.now %>
+
+admin_user:
+  email: admin@test.com
+  confirmed_at: <%= Time.now %>
+  admin: true
```
Comment (yndajas) [inline test/fixtures/users.yml:12]: Would it be better to use test@test.com as the admin user for consistency with the seeds?

**Review by yndajas — CHANGES_REQUESTED** [review]
A few comments but otherwise looks good from my perspective - do you want to review my changes too?

File: app/controllers/admin/imported_commitments_controller.rb (hunk 1 above)
Comment (sergiomarrocoli) [reply app/controllers/admin/imported_commitments_controller.rb:13]: the format/html just renders the view without data, which initialises the vue component so we do need it i think

File: test/fixtures/users.yml (hunk 4 above)
Comment (sergiomarrocoli) [reply test/fixtures/users.yml:12]: changed seeds so admin user is admn@test.com

File: app/controllers/admin/imported_commitments_controller.rb (hunk 2 above)
Comment (sergiomarrocoli) [reply app/controllers/admin/imported_commitments_controller.rb]: changed to authenticate_admin

**Review by yndajas — APPROVED** [review]

**Pull request merged**